provider/aws.go

```diff
@@ -317,6 +318,10 @@ func (p *AWSProvider) records(zones map[string]*route53.HostedZone) ([]*endpoint
 					default:
 						// one of the above needs to be set, otherwise SetIdentifier doesn't make sense
 					}
+
+					if r.HealthCheckId != nil {
```

Comment:
This shouldn't depend on a `SetIdentifier` being set. Can you move that one level out?

Reply:
good catch, @rajatjindal can you move it out of `if r.SetIdentifier != nil ...`? this shouldn't be in the clause.

Reply:
got it. I will make this change

Reply:
I am still testing it, in the meantime, can u confirm if we should check for aliasrecord? as this option is available only if we set aliasrecord to true?

Reply:
requesting feedback from @devkid

Reply:
Not sure what you mean, a health check works with any target, not only alias records.

Reply:
I think @rajatjindal code is correct. Associating a custom health check to a record set in Route53 is only available to records with routing policies other than 'Simple'. Therefore, `HealthCheckId` should only be considered when `SetIdentifier` is also set. @njuettner could you please confirm?